test/Microsoft.AspNet.Mvc.TagHelpers.Test/ScriptTagHelperTest.cs

```diff
@@ -0,0 +1,255 @@
+                Logger = logger,
+            };
+
+            await helper.ProcessAsync(context, output);
+
+            // Assert
+            Assert.NotNull(output.TagName);
+            Assert.False(output.ContentSet);
+        }
+
+        [Fact]
+        public async Task DoesNotRunWhenAllRequiredAttributesAreMissing()
+        {
+            // Arrange
+            var context = MakeTagHelperContext();
+            var output = MakeTagHelperOutput("script");
+            var logger = new Logger();
+
+            // Act
+            var helper = new ScriptTagHelper
+            {
+                Logger = logger,
+            };
+
+            await helper.ProcessAsync(context, output);
+
+            // Assert
+            Assert.NotNull(output.TagName);
+            Assert.False(output.ContentSet);
+        }
+
+        [Fact]
+        public async Task LogsWhenARequiredAttributeIsMissing()
+        {
+            // Arrange
+            var attributes = new Dictionary<string, object>
+                {
+                    { "asp-fallback-src", "http://www.example.com/blank.js" },
+                };
+
+            var context = MakeTagHelperContext(attributes);
+
+            var output = MakeTagHelperOutput("script");
+            var logger = new Logger();
+
+            // Act
+            var helper = new ScriptTagHelper
+            {
+                Logger = logger,
+            };
+
+            await helper.ProcessAsync(context, output);
+
+            // Assert
+            Assert.NotNull(output.TagName);
+            Assert.False(output.ContentSet);
+
+            Assert.Equal(2, logger.Logged.Count);
+
+            Assert.Equal(LogLevel.Warning, logger.Logged[0].Item1);
+            Assert.IsType<MissingAttributeLoggerStructure>(logger.Logged[0].Item2);
+
+            var loggerData0 = (MissingAttributeLoggerStructure)logger.Logged[0].Item2;
+            Assert.Equal(1, loggerData0.MissingAttributes.Count());
+            Assert.Equal("asp-fallback-test", loggerData0.MissingAttributes.Single());
+
+            Assert.Equal(LogLevel.Verbose, logger.Logged[1].Item1);
+            Assert.IsAssignableFrom<ILoggerStructure>(logger.Logged[1].Item2);
+            Assert.StartsWith("Skipping processing for ScriptTagHelper",
+                ((ILoggerStructure)logger.Logged[1].Item2).Format());
+        }
+
+        [Fact]
+        public async Task LogsWhenAllRequiredAttributesAreMissing()
+        {
+            // Arrange
+            var context = MakeTagHelperContext();
+            var output = MakeTagHelperOutput("script");
+            var logger = new Logger();
+
+            // Act
+            var helper = new ScriptTagHelper
+            {
+                Logger = logger,
+            };
+
+            await helper.ProcessAsync(context, output);
+
+            // Assert
+            Assert.NotNull(output.TagName);
+            Assert.False(output.ContentSet);
+
+            Assert.Equal(1, logger.Logged.Count);
```

Comment:
`var log = Assert.Single(logger.Logged)` - You can then use it instead of indexing below